zip-0244.rst

```diff
@@ -1,8 +1,604 @@
+
+   the hash is immediately finalized, without being updated with any
+   additional data
+
+S.2d: txin_sig_digest
+'''''''''''''''''''''
+This is a BLAKE2b-256 hash initialized with the personalization field value
+"Zcash___TxInHash" (3 underscores).
+
+If the signature hash is being computed for a transparent input, the hash
+is updated with the following properties of that input::
+
+   S.2d.i:   prevout     (field encoding)
+   S.2d.ii:  script_code (field encoding)
+   S.2d.iii: value       (8-byte signed little-endian)
+   S.2d.iv:  nSequence   (4-byte unsigned little-endian)
+
+otherwise::
+   
+   the hash is immediately finalized, without being updated with any
+   additional data
+
+S.3: sprout_digest
+``````````````````
+Identical to that specified for the transaction identifier.
+
+S.4: sapling_digest
+```````````````````
+Identical to that specified for the transaction identifier.
+
+Authorizing Data Commitment
+===========================
+
+A new transaction digest algorithm is defined that constructs a digest which commits
+to the authorizing data of a transaction from a tree of BLAKE2b-256 hashes. 
+The overall structure of the hash is as follows:
+
+    auth_digest
+    ├── transparent_scripts_digest
+    ├── sprout_sigs_digest
+    └── sapling_sigs_digest
+
+Each node written as ``snake_case`` in this tree is a BLAKE2b-256 hash of authorizing
+data of the transaction.
+
+The pair (Transaction Identifier, Auth Commitment) constitutes a commitment to all the
+data of a serialized transaction that may be included in a block. 
+
+auth_digest
+-----------
+A BLAKE2b-256 hash of the following values ::
+
+   A1: transparent_scripts_digest (32-byte hash output)
+   A2: sprout_auth_digest         (32-byte hash output)
+   A3: sapling_auth_digest        (32-byte hash output)
+
+The personalization field of this hash is set to::
+
+  "ZTxAuthHash_" || CONSENSUS_BRANCH_ID
+
+A.1: transparent_scripts_digest
+```````````````````````````````
+A BLAKE2b-256 hash of the field encoding of the Bitcoin script associated 
+with each transparent input belonging to the transaction.
+
+The personalization field of this hash is set to::
+
+  "ZTxAuthTransHash"
+
+A.2: sprout_auth_digest
+```````````````````````
+A BLAKE2b-256 hash of the field encoding of the ``zkproof`` values of each
+``JSDescription`` belonging to the transaction, followed by the 
+``joinsplit_pubkey`` and ``joinsplit_sig``::
+
+   A.2a: zkproofs         (field encoding bytes)
+   A.2b: joinsplit_pubkey (field encoding bytes)   
+   A.2b: joinsplit_sig    (field encoding bytes) 
+
+The personalization field of this hash is set to::
+
+  "ZTxAuthSprouHash"
+
+A.3: sapling_auth_digest
+````````````````````````
+A BLAKE2b-256 hash of the field encoding of the Sapling zkproof values
+of each Sapling spend description and output description belonging to 
+the transaction, followed by the field encoding of the binding signature ::
+
+   A.3a: spend_zkproofs  (field encoding bytes)
```

Comment:
I think we need the spend authorization signatures here, because we removed them from T.4a: sapling_spends_digest.

Reply:
Yup, you're correct. I added it to the reference implementation but forgot to update the ZIP: https://github.com/zcash/librustzcash/pull/319/files#diff-3a6e5dddbcd9454863367f5c9532f0359c6af5ddcda50c7cbe17605b9a1012a9R462